### PR TITLE
[v9] Backport Okta OIDC Fixes

### PR DESCRIPTION
**Pull request by zmb3**
Backports #11718

### CODE REVIEW

File: lib/auth/oidc.go [hunk 1]
```diff
@@ -707,19 +694,19 @@ func (a *Server) getClaims(oidcClient *oidc.Client, connector types.OIDCConnecto
...
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 
-	//If the default client secret basic is used the Ping OIDC
-	// will throw an error of multiple client credentials.  Even if you set in Ping
-	// to use Client Secret Post it will return to use client secret basic.
-	// Issue https://github.com/gravitational/teleport/issues/8374
-	if connector.GetProvider() == teleport.Ping {
+	// For OIDC, Ping and Okta will If the default client secret basic is used the Ping OIDC
```
Comment (Tener) [inline lib/auth/oidc.go:702]:
```suggestion
	// For OIDC, Ping and Okta
```

**Review by Tener — APPROVED** [review]
Good to see `createOIDCClient` cleaned up. 

Looks good, except for one comment which sounds broken.

File: lib/auth/oidc.go (hunk 1 above)
Comment (zmb3) [reply lib/auth/oidc.go:702]: Ughh, good catch. Missed this in master :-(

**Review by smallinsky — APPROVED** [review]
Left on question but otherwise LGTM.

File: lib/auth/oidc.go [hunk 2]
```diff
@@ -78,36 +79,22 @@ func (a *Server) createOIDCClient(conn types.OIDCConnector) (*oidc.Client, error
 		return nil, trace.Wrap(err)
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), defaults.WebHeadersTimeout)
-	defer cancel()
-
+	doneSyncing := make(chan struct{})
 	go func() {
-		defer cancel()
+		defer close(doneSyncing)
```
[the remark this answers is not in the document]
Comment (zmb3) [reply lib/auth/oidc.go:84]: Seems like we should, yes. Good find.

This this is not an issue with this specific PR, and this PR is a backport of one that's already been merged, I'll file an issue to address this goroutine leak.

**Pull request merged**